### PR TITLE
BAEL-5978: Case Insensitive Enum Mapping in Spring Boot

### PR DESCRIPTION
**Pull request by azhwani**
Simple PR

### CODE REVIEW

File: spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/enummapping/config/EnumMappingConfig.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.format.FormatterRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+@Configuration
+public class EnumMappingConfig implements WebMvcConfigurer {
+    @Override
+    public void addFormatters(FormatterRegistry registry) {
+        // ApplicationConversionService.configure(registry);
+        // registry.addConverter(new StringToLevelConverter());
```
Comment (JavierPeris) [inline spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/enummapping/config/EnumMappingConfig.java:12]: Why do we have these lines commented?

File: spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/enummapping/enums/Level.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.baeldung.enummapping.enums;
+
+public enum Level {
+
+    LOW, MEDIUM, HIGH;
```
Comment (JavierPeris) [inline spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/enummapping/enums/Level.java:5]: We can remove the semicolon is not necessary.

File: spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/enummapping/editors/LevelEditor.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.apache.commons.lang3.EnumUtils;
+
+import com.baeldung.enummapping.enums.Level;
+
+public class LevelEditor extends PropertyEditorSupport {
+
+    @Override
+    public void setAsText(String text) {
+        setValue(EnumUtils.getEnum(Level.class, text.toUpperCase()));
```
Comment (JavierPeris) [inline spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/enummapping/editors/LevelEditor.java:13]: Shouldn't we check the string is neither null nor empty?

**Review by JavierPeris — COMMENTED** [review]
We should create a new module since this has already 7 articles which is the maximum for a Spring module.

File: spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/enummapping/enums/Level.java (hunk 2 above)
Comment (azhwani) [reply spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/enummapping/enums/Level.java]: done

File: spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/enummapping/editors/LevelEditor.java (hunk 3 above)
Comment (azhwani) [reply spring-boot-modules/spring-boot-basic-customization-2/src/main/java/com/baeldung/enummapping/editors/LevelEditor.java]: Done

Normally, we used _required=false_, so when the parameter isn't specified, the method parameter is bound to null (No converters or property editors are called).

However, I agree. We need to protect our custom converter and property editor against any misuse or NPE

I added additional test cases to take into account _null_ and empty strings

**Review by JavierPeris — APPROVED** [review]

**Pull request merged**